core/physical/include/DiscretizeOperators.h

```diff
@@ -66,8 +69,7 @@ class GPUDownsampleResolution: public PhysicalOperator, public GPUOperator {
 
         }
 
-        Configuration downsampled_configuration() {
-            const auto &base = (*iterator()).configuration();
+        Configuration downsampled_configuration(const Configuration &base) {
             const auto &theta_geometry = get_geometry(Dimension::Theta);
```

Comment:
Is it worth short-circuiting evaluation if the base configuration hasn't changed?

Reply:
I think that's a good idea.